app/scripts/components/exploration/components/map/index.tsx

```diff
@@ -21,60 +18,37 @@ import {
 } from '$components/common/map/controls';
 import MapCoordsControl from '$components/common/map/controls/coords';
 import MapOptionsControl from '$components/common/map/controls/map-options';
-import { projectionDefault } from '$components/common/map/controls/map-options/projections';
-import { useBasemap } from '$components/common/map/controls/hooks/use-basemap';
 import DrawControl from '$components/common/map/controls/aoi';
 import CustomAoIControl from '$components/common/map/controls/aoi/custom-aoi-control';
-import { usePreviousValue } from '$utils/use-effect-previous';
-
-export function ExplorationMap() {
-  const [projection, setProjection] = useState(projectionDefault);
-
-  const {
-    mapBasemapId,
-    setBasemapId,
-    labelsOption,
-    boundariesOption,
-    onOptionChange
-  } = useBasemap();
-
-  const [datasets, setDatasets] = useAtom(timelineDatasetsAtom);
-
-  useReconcileWithStacMetadata(datasets, setDatasets);
-
-  const selectedDay = useAtomValue(selectedDateAtom);
-  const selectedCompareDay = useAtomValue(selectedCompareDateAtom);
-
-  // Different datasets may have a different default projection.
-  // When datasets are selected the first time, we set the map projection to the
-  // first dataset's projection.
-  const prevDatasetCount = usePreviousValue(datasets.length);
-  useEffect(() => {
-    if (!prevDatasetCount && datasets.length > 0) {
-      setProjection(datasets[0].data.projection ?? projectionDefault);
-    }
-  }, [datasets, prevDatasetCount]);
-
-  // If all the datasets are changed through the modal, we also want to update
-  // the map projection, since it is as if the datasets were selected for the
-  // first time.
-  // The only case where we don't want to update the projection is when not all
-  // datasets are changed, because it is not possible to know which projection
-  // to use.
-  const prevDatasetsIds = usePreviousValue(datasets.map((d) => d.data.id));
-  useEffect(() => {
-    if (!prevDatasetsIds) return;
-
-    const newDatasetsIds = datasets.map((d) => d.data.id);
-    const hasSameId = newDatasetsIds.some((id) => prevDatasetsIds.includes(id));
-
-    if (!hasSameId && datasets.length > 0) {
-      setProjection(datasets[0].data.projection ?? projectionDefault);
-    }
-  }, [prevDatasetsIds, datasets]);
-
-  const comparing = !!selectedCompareDay;
+import { BasemapId } from '$components/common/map/controls/map-options/basemap.js';
+
+interface ExplorationMapProps {
+  datasets: TimelineDataset[];
+  selectedDay: Date | null;
+  selectedCompareDay: Date | null;
+  mapBasemapId: BasemapId;
+  setBasemapId: (id: BasemapId) => void;
+  labelsOption: boolean;
+  boundariesOption: boolean;
+  onOptionChange: (option: string, value: boolean) => void;
+  onStyleUpdate: (style: any) => void;
+  projection: ProjectionOptions;
+  setProjection: (projection: ProjectionOptions) => void;
+}
```

Comment:
Typing this might be obsolete now because I kept some of these (ie. the `projection`) within the component (as it was before) instead of passing it as a prop since we're setting the projection based on the passed datasets to the component. Let me know if I'm missing something by not lifting that up to the parent.